Source/WebKit/UIProcess/Cocoa/WebPageProxyCocoa.mm

```diff
@@ -1766,6 +1766,15 @@ void addResult(HashMap<WebCore::FrameIdentifier, Ref<WebCore::LegacyWebArchive>>
 }
 #endif
 
+bool WebPageProxy::shouldInitializeNSAccessibility() const
+{
+#if PLATFORM(MAC)
+    return m_configuration->processPool().hasReceivedAXRequestInUIProcess() || NSWorkspace.sharedWorkspace.isVoiceOverEnabled;
```

Comment:
This doesn't cover for the huge variety of other assistive technologies we have, right? Voice Control, Accessibility Inspector, Full Keyboard Access, Switch Control, more.

Reply:
All of our assistive technology that needs the accessibility tree from WebKit sets AXEnhancedUserInterface, yes. That should cover VoiceOver, Voice Control, Hover Text, etc.

Reply:
There aren't any full-fledged ATs I'm aware of, but there are things like Grammarly. I'm not sure they set AXEnhancedUserInterface, but accessing accessibility attributes of the web page should also enable web accessibility, right?